### PR TITLE
Added bash script and Jupyter notebook

### PR DESCRIPTION
**Pull request by Tanvez**
This PR is to add a bash script to unhide a particular canvas course app in the left hand side nav. This should make it easier to update hidden apps in navs(location is not specified currently) for a set/term of courses. The courses would be manually exported from the canvas data 2 database via a SQL query.
```sql
select id, course_code, enrollment_term_id
from canvas.courses can
where can.tab_configuration like '%{"id":"context_external_tool_<tool id>","hidden":true}%' and enrollment_term_id='<term id>'
```

### CODE REVIEW

**Comment by cmurtaugh** [comment]
My bash is pretty rusty but this looks ok to me!

File: atg_canvas_course_enable_app.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# Check if CSV_FILE, URL, API_KEY, and TOOL_ID are set
+if [ -z "$CSV_FILE" ] || [ -z "$URL" ] || [ -z "$API_KEY" ] || [ -z "$TOOL_ID" ]; then
+    echo "Usage: $0 -f <csv_file> -h <url> -k <api_key> -t <tool_id>"
+    exit 1
+fi
+
+headers=1
+
+
+# Skip the headers if the headers variable is set to 1
```
Comment (jaguillette) [inline atg_canvas_course_enable_app.sh:35]: Why would you want to skip vs. keep the headers? It looks like it's either using or discarding the headers from the CSV file input, so does it depend on how the CSV file is set up?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+
+### Bash script
+If you are opting to use the bash script you will will need to supply the bash command with the following:
+- `-f` with the file path
+- `-h` with the host path similar to the jupyter notebook example
+- `-k` with the api key generated from canvas
+- `-t` with the app tool id(can be found in the canvas data 2 or directly in canvas)
+
+```bash
+bash atg_canvas_course_enable_app.sh -f ../beta_canvas_fall_2024_2025.csv -h 'school.canvas.com' -k 'some-super-secret-key' -t 99999
```
Comment (jaguillette) [inline README.md:19]: Do we have an API key as a secret or SSM param? If so, we could fetch it in the invocation like `-k "$(aws ssm whatever syntax gets the key value)"`. That would be a point in favor of making this a private repo, but could be convenient.

**Review by jaguillette — APPROVED** [review]
Left a couple of comments, but nothing major. Looks good to me! Going to leave comments for my future self in case I get any of the same bright ideas that occurred to me while reviewing, which I then realized were not great.

1. The database credentials don't have write access to the database, and it's a database to get data, not to make changes, so flipping the switch has to be through the API.
2. The query for the courses is better done in the database than in the API, since the API doesn't have a query method to filter based on what's going on in tabs, so it'd be necessary to look at every course and check their tabs if using the API. Not good.

File: atg_canvas_course_enable_app.sh (hunk 1 above)
Comment (Tanvez) [reply atg_canvas_course_enable_app.sh:35]: Yea I was thinking if the CSV didn't have headers there should be a fall back. Depending on how flexible the script can be we can ether add a header variable/input or just assume headers will always be in the csv file.

File: README.md (hunk 2 above)
Comment (Tanvez) [reply README.md:19]: Good call that would make this process more streamlined if we go that route!

**Review by arthurian — APPROVED** [review]
Looks good to me! My bash arg parsing is a bit rusty too, but the core parts look right.  👍

**Pull request merged**